Sources/DeepLearning/Layer.swift

```diff
@@ -1299,11 +1299,16 @@ public struct SimpleRNNCell<Scalar: TensorFlowFloatingPoint>: RNNCell, VectorNum
         return TensorShape([1, weight.shape[1]])
     }
 
-    public var zeroState: Tensor<Scalar> {
-        return Tensor(zeros: stateShape)
+    public var zeroState: State {
+        return State(state: Tensor(zeros: stateShape))
+    }
+
+    // TODO(TF-507): Revert to `typealias State = Tensor<Scalar>` after
+    // SR-10697 is fixed.
+    public struct State: Differentiable {
+        let state: Tensor<Scalar>
```

Comment:
1. Could you call this `value` so that it doesn't repeat the struct name?
2. Since this is public API, please make the property public and add a memberwise initializer that has no argument labels.

Reply:
Done.